OPHD/States/MapViewStateUi.cpp

```diff
@@ -24,6 +24,16 @@
 using namespace constants;
 
 
+static void setOverlay(Button& button, TileList& tileList, Tile::Overlay overlay)
+{
+	auto overlayToUse = button.toggled() ? overlay : Tile::Overlay::None;
+	for (auto tile : tileList)
+	{
+		tile->overlay(overlayToUse);
+	}
+}
```

Comment:
You could potentially pass a `boolean` instead of a `Button&` (which could potentially have been a `const Button&`).

By that token, you could also have passed just the `overlay`, and determined it's value outside this method. Though that starts to duplicate code, so I can see why you'd have done it this way.

----

What happens if a tile has two overlays set? Or rather, if the `tileList` collections have non-overlapping tiles, and you switch from one overlay to another? does it leave behind ghostly highlights from the old overlay?

Does the code need to have old overlays unset? Can it support multiple overlays at the same time?

Reply:
The code has old overlays unset. Only one overlay can be set at a time. It's not the most efficient method but it's simple and does the job.

I opted to use Button& for future overlays.